lib/api/multipartDelete.js

```diff
@@ -54,17 +55,25 @@ function multipartDelete(authInfo, request, log, callback) {
                         metadataValMPUparams.requestType =
                             'bucketPolicyGoAhead';
                     }
-                    return next();
+                    const locationConstraint =
+                        destinationBucket.getLocationConstraint();
+                    return next(null, locationConstraint);
                 });
         },
-        function checkMPUval(next) {
+        function checkMPUval(locationConstraint, next) {
             services.metadataValidateMultipart(metadataValParams,
                 (err, mpuBucket, mpuOverviewArray) => {
                     if (err === errors.NoSuchUpload) {
                         log.trace('did not find valid mpu with uploadId' +
                         `${uploadId}`, { method: 'multipartDelete' });
-                        // TODO in rel/6.3 and later: return 404 if
-                        // usEastBehavior is enabled in config
+                        // if legacy behavior is enabled for 'us-east-1' and
+                        // request is from 'us-east-1', return 404 instead of
+                        // 204
+                        if (config.usEastBehavior &&
+                            locationConstraint === 'us-east-1') {
+                            return callback(err);
+                        }
+                        // otherwise ignore error and return 204 status code
                         return callback();
```

Comment:
It's interesting that we don't return error in the callback if it's any error other than `errors.NoSuchUpload`. We have to fix this, may be in a different PR.